src/approx_inverse_retraction.jl

```diff
@@ -0,0 +1,104 @@
+"""
+    ApproximateInverseRetraction <: AbstractInverseRetractionMethod
+
+An abstract type for representing approximate inverse retraction methods.
+"""
+abstract type ApproximateInverseRetraction <: AbstractInverseRetractionMethod end
```

Comment:
Perhaps it's a confusion on my part. Aren't most of our inverse retractions exact inverses of some retraction within some radius? Because that is the sense in which this an approximate inverse retraction. The idea is that you may be able to write some retraction in closed-form but not its inverse, and subtypes of this type would be general-purpose approaches to approximately invert that retraction. Which I guess would end up being a first-order approximation to the log, but indirectly.

Reply:
This is also basically a generalization of geodesic shooting to not just the exponential but also any retraction.

Reply:
ah so its a general approximation method, ten its ok, for sure.